File: LeetCode/Easy/0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowel = set(["a","e","i","o","u", "A", "E", "I", "O", "U"])
        i = 0
        j = len(s) - 1
        length = len(s)
        ret = list(s)
        while i < j:
            while i < length - 1 and s[i] not in vowel:
                i += 1
            while j > 0 and s[j] not in vowel:
                j -= 1
            if i < j:
                ret[j] = s[i]
                ret[i] = s[j]
            i += 1
            j -= 1
        return "".join(ret)

File: LeetCode/Easy/0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        unordered_set<char> vowel = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        int i = 0;
        int j = s.length() - 1;
        vector<char> ret(s.begin(), s.end());

        
        while (i < j){
            while (i < j && vowel.find(s[i]) == vowel.end()) {
                ret[i] = s[i];
                ++i;
            }
            while (i < j && vowel.find(s[j]) == vowel.end()) {
                ret[j] = s[j];
                --j;
            }
            if (i < j) {
                ret[i] = s[j];
                ret[j] = s[i];
                ++i;
                j--;
            }
        }

        string result(ret.begin(), ret.end());
        return result;
    }
};